### PR TITLE
WIP 1.20

### PR DESCRIPTION
**Pull request by Big-Iron-Cheems**
## Type of change

- [ ] Bug fix
- [X] New feature

## Description

WIP 1.20 update.

## Related issues

Mention any issues that this pr relates to.
#3707 

# How Has This Been Tested?
Apart from success during compile and build phase, this was not yet tested in-game.
Waiting for Baritone to update to 1.20 before further testing is possible.

# Checklist:

- [X] My code follows the style guidelines of this project.
- [X] I have added comments to my code in more complex areas.
- [ ] I have tested the code in both development and production environments.

### CODE REVIEW

**Comment by sebseb7** [comment]
[08/06/2023 13:10:22 PM] java.lang.RuntimeException: Mixin transformation of net.minecraft.class_338 failed
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427)
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112)
[08/06/2023 13:10:22 PM] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
[08/06/2023 13:10:22 PM] 	at net.minecraft.class_315.<init>(class_315.java:469)
[08/06/2023 13:10:22 PM] 	at net.minecraft.class_310.<init>(class_310.java:460)
[08/06/2023 13:10:22 PM] 	at net.minecraft.client.main.Main.main(Main.java:211)
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468)
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
[08/06/2023 13:10:22 PM] Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
[08/06/2023 13:10:22 PM] 	at meteordevelopment.meteorclient.asm.Asm$Transformer.transformClassBytes(Asm.java:102)
[08/06/2023 13:10:22 PM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422)
[08/06/2023 13:10:22 PM] 	... 10 more
[08/06/2023 13:10:22 PM] Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector onMessageIndicator(Lnet/minecraft/class_303$class_7590;)Lnet/minecraft/class_7591; in meteor-client.mixins.json:ChatHudMixin from mod meteor-client failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap meteor-client-refmap.json
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1384)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1062)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:402)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:327)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
[08/06/2023 13:10:22 PM] 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
[08/06/2023 13:10:22 PM] 	... 14 more

**Comment by Big-Iron-Cheems** [comment]
Thanks for the quick report, found the issue, fix will be coming shortly 👍

**Comment by PoTheMagicDragon** [comment]
I just tried building and running it with Prism Launcher and java 17 and got this on startup. 

TLDR: 
Description: Initializing game
java.lang.ExceptionInInitializerError
`
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [meteor-client.mixins.json:DrawContextMixin from mod meteor-client] from phase [DEFAULT] in config [meteor-client.mixins.json] FAILED during APPLY
`
& 
`
Caused by: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Critical injection failure: @Inject annotation on onComponentConstruct could not find any targets matching 'drawTooltip(Lnet/minecraft/client/font/TextRenderer;Ljava/util/List;Ljava/util/Optional;II)V' in net.minecraft.class_332. Using refmap meteor-client-refmap.json [PREINJECT Applicator Phase -> meteor-client.mixins.json:DrawContextMixin from mod meteor-client -> Prepare Injections ->  -> handler$zda000$meteor-client$onComponentConstruct(Lnet/minecraft/class_327;Ljava/util/List;Ljava/util/Optional;IILorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V -> Parse]
`

https://mclo.gs/KePcGWv

**Comment by PoTheMagicDragon** [comment]
Nice! Looks like that last one was fixed. This one may be my own ignorance, but I now get this error on startup:

`[06:03:44] [main/WARN]: Error loading class: baritone/command/defaults/ComeCommand (java.lang.ClassNotFoundException: baritone/command/defaults/ComeCommand)`

`[06:03:44] [main/WARN]: @Mixin target baritone.command.defaults.ComeCommand was not found meteor-client-baritone.mixins.json:ComeCommandMixin from mod meteor-client`

`[06:03:44] [main/WARN]: Error loading class: baritone/api/Settings$Setting (java.lang.ClassNotFoundException: baritone/api/Settings$Settings`

https://mclo.gs/5SCuZfu

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
Comment (CDAGaming) [inline gradle/wrapper/gradle-wrapper.properties:3]: Why are you updating random shit for an MC version update.
Please put this type of stuff in different PRs (Same goes for the indent changes in several files, dont do that)

File: src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java [hunk 2]
```diff
@@ -318,10 +320,7 @@ public void removed() {
             Input.setCursorStyle(CursorStyle.Default);
 
             loopWidgets(root, widget -> {
-                if (widget instanceof WTextBox textBox) {
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java:321]: While the cleanup is nice, please keep this to changes actually relevant to 1.20

File: src/main/java/meteordevelopment/meteorclient/mixin/ChatHudMixin.java [hunk 3]
```diff
@@ -163,22 +167,20 @@ private static double getMessageOpacityMultiplier(int age) {
 
     private void drawIcon(MatrixStack matrices, String line, int y, float opacity) {
         if (METEOR_PREFIX_REGEX.matcher(line).find()) {
-            RenderSystem.setShaderTexture(0, METEOR_CHAT_ICON);
             matrices.push();
             RenderSystem.setShaderColor(1, 1, 1, opacity);
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/mixin/ChatHudMixin.java:171]: This probably is no longer needed, since drawContext is meant to take over a lot of the RenderSystem calls

File: src/main/java/meteordevelopment/meteorclient/mixin/ClientPlayerEntityMixin.java [hunk 4]
```diff
@@ -73,7 +73,8 @@ private void onPushOutOfBlocks(double x, double d, CallbackInfo info) {
 
     @Inject(method = "damage", at = @At("HEAD"))
     private void onDamage(DamageSource source, float amount, CallbackInfoReturnable<Boolean> info) {
-        if (Utils.canUpdate() && world.isClient && canTakeDamage()) MeteorClient.EVENT_BUS.post(DamageEvent.get(this, source, amount));
+        if (Utils.canUpdate() && getWorld().isClient && canTakeDamage())
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/mixin/ClientPlayerEntityMixin.java:76]: Same thing as prior comment, please do not do cleanups that are un-related to 1.20 -- store in seperate PR

File: src/main/java/meteordevelopment/meteorclient/mixin/InGameHudMixin.java [hunk 5]
```diff
@@ -40,7 +40,7 @@ public abstract class InGameHudMixin {
     @Shadow public abstract void clear();
 
     @Inject(method = "render", at = @At("TAIL"))
-    private void onRender(MatrixStack matrixStack, float tickDelta, CallbackInfo info) {
+    private void onRender(DrawContext context, float tickDelta, CallbackInfo ci) {
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/mixin/InGameHudMixin.java:43]: Why rename the CallbackInfo?

File: src/main/java/meteordevelopment/meteorclient/systems/modules/combat/SelfTrap.java [hunk 6]
```diff
@@ -182,29 +182,29 @@ private void findPlacePos(Block block) {
         BlockPos pos = mc.player.getBlockPos();
 
         switch (topPlacement.get()) {
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/combat/SelfTrap.java:184]: Why are we switching to enhanced switch casing, this is also not necessary for the 1.20 port

File: src/main/java/meteordevelopment/meteorclient/utils/Utils.java [hunk 7]
```diff
@@ -349,71 +352,71 @@ public static String titleToName(String title) {
     }
 
     public static String getKeyName(int key) {
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/utils/Utils.java:354]: Absolutly not. Please stop making un-related changes to the 1.20 port

File: src/main/java/meteordevelopment/meteorclient/utils/player/SlotUtils.java [hunk 8]
```diff
@@ -36,26 +37,26 @@ public static int indexToId(int i) {
         ScreenHandler handler = mc.player.currentScreenHandler;
 
         if (handler instanceof PlayerScreenHandler) return survivalInventory(i);
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/utils/player/SlotUtils.java:39]: This can break if you just yeet all the `else if`
Same comment as above, unneeded change

File: src/main/java/meteordevelopment/meteorclient/utils/player/SlotUtils.java [hunk 9]
```diff
@@ -134,17 +136,17 @@ private static int hopper(int i) {
     private static int horse(ScreenHandler handler, int i) {
         AbstractHorseEntity entity = ((HorseScreenHandlerAccessor) handler).getEntity();
 
-        if (entity instanceof LlamaEntity) {
-            int strength = ((LlamaEntity) entity).getStrength();
+        if (entity instanceof LlamaEntity llamaEntity) {
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/utils/player/SlotUtils.java:139]: not needed

File: src/main/java/meteordevelopment/meteorclient/utils/render/PeekScreen.java [hunk 10]
```diff
@@ -59,22 +59,21 @@ public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
...
         }
         return false;
     }
 
     @Override
-    protected void drawBackground(MatrixStack matrices, float delta, int mouseX, int mouseY) {
+    protected void drawBackground(DrawContext context, float delta, int mouseX, int mouseY) {
         Color color = Utils.getShulkerColor(storageBlock);
 
         RenderSystem.setShader(GameRenderer::getPositionTexProgram);
```
Comment (CDAGaming) [inline src/main/java/meteordevelopment/meteorclient/utils/render/PeekScreen.java:72]: Both RenderSystem calls here are likely unnecesary

**Review by CDAGaming — CHANGES_REQUESTED** [review]
Lots of unnecesary changes are in this PR that appear to be un-related to the 1.20 port itself. I am too busy at this time to make this myself, so I would recommend one of the contributors to further review this before thinking of merging this PR.

File: build.gradle [hunk 11]
```diff
@@ -45,21 +45,21 @@ configurations {
...
     modCompileOnly fabricApi.module("fabric-renderer-indigo", project.fapi_version)
     modCompileOnly("maven.modrinth:sodium:${project.sodium_version}") { transitive = false }
     modCompileOnly("maven.modrinth:lithium:${project.lithium_version}") { transitive = false }
     modCompileOnly("maven.modrinth:iris:${project.iris_version}") { transitive = false }
     //modCompileOnly("io.vram:canvas-fabric-mc119:1.0.+") { transitive = false } // TODO: 1.19.3
 
-	// Baritone (https://github.com/MeteorDevelopment/baritone)
-    modInclude "baritone:fabric:${project.minecraft_version}-SNAPSHOT"
+    // Baritone (https://github.com/MeteorDevelopment/baritone)
+    modInclude "baritone:fabric:1.19.4-SNAPSHOT"
```
Comment (BDARmech) [inline build.gradle]: Going back in version?

Comment (BDARmech) [inline build.gradle]: https://github.com/MeteorDevelopment/baritone is 1.2

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (BDARmech) [reply gradle/wrapper/gradle-wrapper.properties:3]: I jusr like the \:/
